### PR TITLE
Allow starting voila from a Voila instance

### PR DESCRIPTION
**Pull request by timkpaine**
The following should now work without error:
```python3
from voila.app import Voila
v = Voila()
v.notebook_path = 'bqplot.ipynb'
v.start()
```

### CODE REVIEW

**Comment by SylvainCorlay** [comment]
Rebased the PR.

**Comment by timkpaine** [comment]
@SylvainCorlay thanks!

File: voila/app.py [hunk 1]
```diff
@@ -133,24 +166,24 @@ def _default_root_dir(self):
 
     def parse_command_line(self, argv=None):
         super(Voila, self).parse_command_line(argv)
-        self.notebook_path = self.extra_args[0] if len(self.extra_args) == 1 else None
-        self.nbconvert_template_paths = []
-        self.template_paths = []
-        self.static_paths = [self.static_root]
+
+    def setup_template_dirs(self):
+        self.notebook_path = self.notebook_path if self.notebook_path else self.extra_args[0] if len(self.extra_args) == 1 else None
```
[the remark this answers is not in the document]
Comment (timkpaine) [reply voila/app.py]: this should already be setup from the default values of the new attributes

**Comment by timkpaine** [comment]
@maartenbreddels let me try to change the order of templates

**Comment by timkpaine** [comment]
@maartenbreddels does ` [06c84fd](/QuantStack/voila/pull/61/commits/06c84fd9dc5b56e0acee688f2a2955b43f717c0a)` look ok for inserting the paths in the proper reverse order, but starting after any user-provided paths (in order)?

**Comment by maartenbreddels** [comment]
I had the same idea, but that messes up the order of static_paths. I think as it was is ok. The proper way to customize the paths, is to set template=None, e.g.:
```
python -m voila notebooks/basics.ipynb \
  "--Voila.nbconvert_template_paths=['share/jupyter/voila/template/gridstack/nbconvert_templates', 'share/jupyter/voila/template/default/nbconvert_templates']" \
  "--Voila.template_paths=['share/jupyter/voila/template/default/templates']" \
  --Voila.log_level=DEBUG --template=None
```

In you revert the last commit, I can merge this. There still is an issue in master that cause the above not to work, I'll fix that separately.

**Comment by maartenbreddels** [comment]
I'll create a new branch since I have to add in a few more things, I'll pick it up from there.

**Pull request closed without merging**